src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+public class FigureSupplier {
+    public static final Color WHITE = Color.WHITE;
+    public static final int DEFAULT_FIGURE_RADIUS = 10;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int type = (int) (Math.random() * 5);
+        Color color = colorSupplier.getRandomColor();
+
+        switch (type) {
+            case 0:
+                double side = Math.random() * 10 + 1;
+                return new Square(color, side);
+            case 1:
+                double width = Math.random() * 10 + 1;
+                double height = Math.random() * 10 + 1;
+                return new Rectangle(color, width, height);
+            case 2:
+                double firstLeg = Math.random() * 10 + 1;
+                double secondLeg = Math.random() * 10 + 1;
+                return new RightTriangle(color, firstLeg, secondLeg);
+            case 3:
+                double radius = Math.random() * 10 + 1;
+                return new Circle(color, radius);
+            case 4:
+                double topBase = Math.random() * 10 + 1;
+                double bottomBase = Math.random() * 10 + 1;
+                double height2 = Math.random() * 10 + 1;
+                return new IsoscelesTrapezoid(color, topBase, bottomBase, height2);
+            default:
+                return null;
```

Comment:
It is better to return the default value
```suggestion
                return getDefaultFigure();
```